LSPS0/README.md

```diff
@@ -0,0 +1,606 @@
+> description of this error, which developers of clients can
+> then use to help guide the evolution of the specification, or
+> to comply with later versions of the LSPS spec.
+
+#### Disconnection Handling
+
+Networks are unreliable, and network participants are also unreliable.
+
+Clients MUST use a high-entropy `id` string for JSON-RPC 2.0 requests,
+such as a UUID, or a hex encoding of a random binary blob of at least
+80 bits, with randomness acquired from a cryptographically secure
+source.
+Clients MUST NOT use just a simple incrementing counter for the `id`.
+
+If a client receives a JSON-RPC 2.0 response with an `id` it does
+not remember sending a request for, it MUST ignore that response.
+
+> **Rationale** Clients, LSPs, and the network between them
+> can individually be unreliable, leading to clients that forget
+> `id`s they issued previously, or clients or LSPs thinking that
+> the other side may have restarted when it is the network
+> between them that failed.
+> Thus, random `id`s picked from a large space are the safest
+> when the client might lose any counter state (by crashing if
+> using an in-memory counter, or loss of persistent storage on
+> hardware without storage redundancy), and are resilient
+> against reconnections when both sides remained running.
+
+Clients MAY include additional information in their `id` for
+internal tracking, as long as the total `id` has sufficient
+entropy for universal uniqueness.
+
+Clients SHOULD internally impose a reasonable timeout, on
+the scale of minutes, for receiving a response for a request,
+and SHOULD treat a timeout event as a temporary server failure,
+and forget the `id` of the timed-out request.
+
+> **Rationale** The LSP might crash between the time the client
+> makes the request to the time it could complete processing of
+> the response, and thus lost track of the request.
+
+If the LSP is unable to deliver a response to the client due
+to a disconnection, it SHOULD treat this no differently from
+successfully delivering the response but the client then does
+not "follow up" on any action after that.
+
+> **Rationale** Even if the LSP delivers the response to the
+> client, the client could then crash while processing the
+> response, which means that the LSP cannot be sure that any
+> response is received by the client anyway.
+
+### Lightning Feature Bit
+
+The [BOLT7][] specification describes the `node_announcement`
+gossip message, which includes a `features` bit field.
+The [BOLT1][] specification describes the `init` message,
+which also includes a `features` bit field.
+
+[BOLT7]: https://github.com/lightning/bolts/blob/f7dcc32694b8cd4f3a1768b904f58cb177168f29/07-routing-gossip.md
+[BOLT1]: https://github.com/lightning/bolts/blob/f7dcc32694b8cd4f3a1768b904f58cb177168f29/01-messaging.md
+
+LSPs MUST set the `features` bit numbered 729
+(`option_supports_lsps`) in both the `init` message on connection
```

Comment:
Maybe add a note that you should set that feature bit when you implement any of the lsp specifications (you don't have to implement all), because there is the `listprotocols` call to distinguish between them.

Reply:
Done